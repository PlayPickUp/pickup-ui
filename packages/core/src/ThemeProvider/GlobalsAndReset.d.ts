import { ReactNode } from "react";
import { DefaultTheme } from "../../../../types";
declare const GlobalsAndReset: import("react").ComponentType<
	Pick<
		{
			children: ReactNode | any;
		},
		"children"
	> &
		Partial<
			import("react-jss").WithStylesProps<
				(
					theme: DefaultTheme
				) => {
					"@global": {
						"html, body, div, span, applet, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, a, abbr, acronym, address, big, cite, code, del, dfn, em, img, ins, kbd, q, s, samp, small, strike, strong, sub, sup, tt, var, b, u, i, center, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, canvas, details, embed, figure, figcaption, footer, header, hgroup, menu, nav, output, ruby, section, summary, time, mark, audio, video": {
							margin: string;
							padding: string;
							border: string;
							fontSize: string;
							font: string;
							verticalAlign: string;
						};
						"article, aside, details, figcaption, figure, footer, header, hgroup, menu, nav, section": {
							display: string;
						};
						body: {
							lineHeight: string;
						};
						"ol, ul": {
							listStyle: string;
						};
						"blockquote, q": {
							quotes: string;
						};
						"blockquote:before, blockquote:after, q:before, q:after": {
							content: string;
						};
						table: {
							borderCollapse: string;
							borderSpacing: string;
						};
						"html, body": {
							width: string;
							minHeight: string;
							fontFamily: string;
							fontSize: import("react").ReactText;
							fontWeight: number;
							color: string;
							backgroundColor: string;
							"-webkit-font-smoothing": string;
							"-moz-osx-font-smoothing": string;
							margin: string;
							padding: string;
						};
						html: {
							boxSizing: string;
							background: string;
						};
						"*, *:before, *:after": {
							boxSizing: string;
						};
						button: {
							display: string;
							"-webkit-appearance": string;
							"-moz-appearance": string;
							userSelect: string;
							textDecoration: string;
							fontFamily: string;
							fontSize: string;
							color: string;
							cursor: string;
							border: string;
							backgroundColor: string;
							borderColor: string;
							padding: string;
							margin: string;
							"&:disabled": {
								cursor: string;
							};
						};
					};
				}
			>
		> & {
			innerRef?:
				| import("react").RefObject<any>
				| ((instance: any) => void)
				| undefined;
		}
>;
export default GlobalsAndReset;
//# sourceMappingURL=GlobalsAndReset.d.ts.map
