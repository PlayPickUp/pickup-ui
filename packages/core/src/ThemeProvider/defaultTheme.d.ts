import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
