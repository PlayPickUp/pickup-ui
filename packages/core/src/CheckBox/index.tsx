import React from "react";
import { createUseStyles } from "react-jss";
import { DefaultTheme, CheckBoxProps } from "../types";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  contain: {
    display: "block",
    position: "relative",
    cursor: "pointer",
    fontSize: "22px",
    userSelect: "none",
    height: 16,
    "& input": {
      /*  Hide the browser's default checkbox  */
      position: "absolute",
      opacity: 0,
      cursor: "pointer",
      height: 0,
      width: 0,
      /* Show the checkmark when checked */
      "&:checked": {
        "& ~ $checkmark": {
          /* When the checkbox is checked*/
          backgroundColor: theme.colors.primary.base,
          "&:after": {
            boxSizing: "border-box",
            display: "block",
          },
        },
      },
    },
    "& $checkmark": {
      "&:after": {
        left: "5px",
        top: "2px",
        width: "5px",
        height: "10px",
        border: "solid white",
        borderWidth: "0 3px 3px 0",
        transform: "rotate(45deg)",
      },
    },
    /* On mouse-over, add a background color */
    "&:hover": {
      "& input": {
        "& ~ $checkmark": {
          backgroundColor: theme.colors.primary.lightBase,
        },
      },
    },
  },
  checkBoxLabel: {
    verticalAlign: "middle",
    position: "relative",
    left: `${theme.spacing.base * 6}px`,
    bottom: `${theme.spacing.base}px`,
    fontSize: 15,
  },
  checkmark: {
    position: "absolute",
    top: 0,
    left: 0,
    height: "16px",
    width: "16px",
    borderRadius: "3px",
    backgroundColor: theme.colors.primary.base,
    "&:after": {
      content: '""',
      position: "absolute",
      display: "none",
    },
  },
}));

const CheckBox: React.FC<CheckBoxProps> = ({
  checked = false,
  onChange,
  label,
  ...props
}) => {
  const classes = useStyles();
  const [isChecked, setIsChecked] = React.useState<boolean>(checked);

  const handleChange = () => {
    setIsChecked(!isChecked);
    onChange(!isChecked);
  };

  return (
    <>
      <label className={classes.contain} htmlFor={props.name}>
        <input
          data-testid="checkbox"
          type="checkbox"
          checked={isChecked}
          onChange={handleChange}
        />
        <span className={classes.checkmark} />
        {label && <label className={classes.checkBoxLabel}>{label}</label>}
      </label>
    </>
  );
};

export default CheckBox;
