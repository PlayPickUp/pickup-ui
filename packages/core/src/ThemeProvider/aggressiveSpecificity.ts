/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
