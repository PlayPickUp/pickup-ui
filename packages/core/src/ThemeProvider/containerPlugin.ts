/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
