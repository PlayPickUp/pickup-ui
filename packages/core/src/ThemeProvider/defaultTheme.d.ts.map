{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}