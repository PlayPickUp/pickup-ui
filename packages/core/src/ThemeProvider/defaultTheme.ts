import {
  Breakpoints,
  CreateShadow,
  DefaultGradients,
  DefaultShadows,
  DefaultSpacing,
  DefaultTheme,
  DefaultThemeColors,
  DefaultTypography,
  DefaultZIndex,
  FontFamilies,
  GreenMap,
  GreyMap,
  MediaQuery,
  PurpleMap,
  RedMap,
  ResponsiveFonts,
} from "../types";

// Color me PickUp âš¡
const purple: PurpleMap = {
  light: "#F5F0FF",
  lightBase: "#AFA0FF",
  base: "#7A60FF",
  dark: "#5E24FF",
  transparent: "#EAE5FF",
};

const green: GreenMap = {
  light: "#EAFDF7",
  base: "#25DA95",
};

const grey: GreyMap = {
  light: "#F3F2F5",
  lightBase: "#DDDDDD",
  base: "#9D9D9E",
  dark: "#615E66",
  medium: "#DFDAE6",
};

const red: RedMap = {
  light: "#FFF1F4",
  base: "#FF1244",
};

const colors: DefaultThemeColors = {
  primary: { ...purple },
  secondary: { ...grey },
  green,
  purple,
  grey,
  red,
  black: "#2C2933",
  white: "#FFF",
  success: green.base,
  error: red.base,
};

// Font and Typography
const fontFamilies: FontFamilies = {
  body: "'Inter', Helvetica, sans-serif",
  headline: "'Ringside-Condensed', Helvetica, sans-serif",
};

const fontStyles: ResponsiveFonts = {
  desktop: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 48,
      lineHeight: "54px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 36,
      lineHeight: "40px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "28px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "22px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "16px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 18,
      lineHeight: "30px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 13,
      lineHeight: "20px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
  mobile: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 34,
      lineHeight: "38px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "26px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "15px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 11,
      lineHeight: "13px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 15,
      lineHeight: "24px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 12,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
};

const typography: DefaultTypography = {
  fontFamilies,
  fontStyles,
};

// Fancy ass Material UI inspired shadows
const shadowKeyUmbraOpacity = 0.2;
const shadowKeyPenumbraOpacity = 0.14;
const shadowAmbientShadowOpacity = 0.12;

const createShadow: CreateShadow = (...px) => {
  return [
    `${px[0]}px ${px[1]}px ${px[2]}px ${px[3]}px rgba(0, 0, 0, ${shadowKeyUmbraOpacity})`,
    `${px[4]}px ${px[5]}px ${px[6]}px ${px[7]}px rgba(0, 0, 0, ${shadowKeyPenumbraOpacity})`,
    `${px[8]}px ${px[9]}px ${px[10]}px ${px[11]}px rgba(0, 0, 0, ${shadowAmbientShadowOpacity})`,
  ].join(",");
};

const shadows: DefaultShadows = [
  createShadow(0, 1, 3, 0, 0, 1, 1, 0, 0, 2, 1, -1),
  createShadow(0, 1, 5, 0, 0, 2, 2, 0, 0, 3, 1, -2),
  createShadow(0, 1, 8, 0, 0, 3, 4, 0, 0, 3, 3, -2),
  createShadow(0, 2, 4, -1, 0, 4, 5, 0, 0, 1, 10, 0),
  createShadow(0, 3, 5, -1, 0, 5, 8, 0, 0, 1, 14, 0),
  createShadow(0, 3, 5, -1, 0, 6, 10, 0, 0, 1, 18, 0),
  createShadow(0, 4, 5, -2, 0, 7, 10, 1, 0, 2, 16, 1),
  createShadow(0, 5, 5, -3, 0, 8, 10, 1, 0, 3, 14, 2),
  createShadow(0, 5, 6, -3, 0, 9, 12, 1, 0, 3, 16, 2),
  createShadow(0, 6, 6, -3, 0, 10, 14, 1, 0, 4, 18, 3),
  createShadow(0, 6, 7, -4, 0, 11, 15, 1, 0, 4, 20, 3),
  createShadow(0, 7, 8, -4, 0, 12, 17, 2, 0, 5, 22, 4),
  createShadow(0, 7, 8, -4, 0, 13, 19, 2, 0, 5, 24, 4),
  createShadow(0, 7, 9, -4, 0, 14, 21, 2, 0, 5, 26, 4),
  createShadow(0, 8, 9, -5, 0, 15, 22, 2, 0, 6, 28, 5),
  createShadow(0, 8, 10, -5, 0, 16, 24, 2, 0, 6, 30, 5),
  createShadow(0, 8, 11, -5, 0, 17, 26, 2, 0, 6, 32, 5),
  createShadow(0, 9, 11, -5, 0, 18, 28, 2, 0, 7, 34, 6),
  createShadow(0, 9, 12, -6, 0, 19, 29, 2, 0, 7, 36, 6),
  createShadow(0, 10, 13, -6, 0, 20, 31, 3, 0, 8, 38, 7),
  createShadow(0, 10, 13, -6, 0, 21, 33, 3, 0, 8, 40, 7),
  createShadow(0, 10, 14, -6, 0, 22, 35, 3, 0, 8, 42, 7),
  createShadow(0, 11, 14, -7, 0, 23, 36, 3, 0, 9, 44, 8),
  createShadow(0, 11, 15, -7, 0, 24, 38, 3, 0, 9, 46, 8),
];

// Base Unit Spacings
const spacing: DefaultSpacing = {
  base: 4, // ðŸš¨ changing this value (4) could have VERY BAD CONCEQUENCES! BE VERY SURE, MAYBE YOU JUST NEED TO ADD A VALUE? ðŸ¤”
};

// Default zIndex values
const zIndex: DefaultZIndex = {
  overlay: 999,
};

// Default Border Radius
const borderRadius = 4;

// Media Queries
const mediaQuery: MediaQuery = (minWidth: number) => {
  return `@media (min-width: ${minWidth}px)`;
};

// Breakpoints
const breakpoints: Breakpoints = {
  small: 787,
  medium: 1029,
  large: 1280,
};

// Default Gradient Values
const gradients: DefaultGradients = {
  main: `linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%)`,
  hover: `linear-gradient(112.25deg, #914EFF 35.38%, #5D1BFF 122.56%)`,
  disabled:
    "linear-gradient(112.25deg, rgba(145,78,255,0.2) -20.38%, rgba(93,27,255,0.2) 122.56%)",
};

export const defaultTheme: DefaultTheme = {
  borderRadius,
  breakpoints,
  colors,
  gradients,
  mediaQuery,
  shadows,
  spacing,
  typography,
  zIndex,
};
