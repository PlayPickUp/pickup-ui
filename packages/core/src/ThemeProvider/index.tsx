import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
