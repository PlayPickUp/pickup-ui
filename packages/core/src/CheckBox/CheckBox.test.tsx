import React from "react";
import { render, fireEvent } from "@testing-library/react";

import ThemeProvider from "../ThemeProvider";
import CheckBox from ".";

test("Renders without crashing, matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <CheckBox
        onChange={(internal: boolean) => {
          console.log(internal);
        }}
      />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("CheckBox can be Checked, and prop can be passed", () => {
  const { getByTestId } = render(
    <ThemeProvider>
      <CheckBox
        checked={true}
        onChange={(internal: boolean) => {
          console.log(internal);
        }}
      />
    </ThemeProvider>
  );
  const input = getByTestId("checkbox");
  expect(input.checked).toEqual(true);
  fireEvent.click(input);
  expect(input.checked).toEqual(false);
});
