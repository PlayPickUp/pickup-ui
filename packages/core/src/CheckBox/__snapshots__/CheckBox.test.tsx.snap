// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j0 {
  cursor: pointer;
  height: 16px;
  display: block;
  position: relative;
  font-size: 22px;
  user-select: none;
}

body #PickUpUI .j1 {
  top: 0;
  left: 0;
  width: 16px;
  height: 16px;
  position: absolute;
  border-radius: 3px;
  background-color: #7A60FF;
}

<div>
  <div
    id="PickUpUI"
  >
    <label
      class="j0"
    >
      <input
        data-testid="checkbox"
        type="checkbox"
      />
      <span
        class="j1"
      />
    </label>
  </div>
</div>
`;
